test/suites/depend.bash

```diff
@@ -294,6 +294,29 @@ EOF
     expect_stat 'files in cache' 9
 
     # -------------------------------------------------------------------------
+    TEST "__DATE__ in header file results in direct cache hit as the date remains the same"
```

Comment:
This functionality is not specific to the depend mode, so please move it to direct.bash instead.